### DISCUSSION TITLE
Foreach with GetCurrentAsync

### DISCUSSION THREAD

**Discussion by IvanKonov** [dotnet/csharplang#839] [General]
Consider the possibility of an asynchronous enumeration of collections. For example, through the support of the GetCurrentAsync method in foreach.

**Comment by orthoxerox** [comment]
See #43

**Comment by DavidArno** [comment]
Also, from #840, see https://channel9.msdn.com/Blogs/Seth-Juarez/A-Preview-of-C-8-with-Mads-Torgersen, where Mads discusses possible syntax and functionality for this feature, possibly coming to C# 8.
